File: filesystem/simple/csharp/ReadVersion.cs
// build:
//  csc ReadVersion.cs

using System;
using System.IO;

class ReadVersion
{
	public static void Usage()
	{
		Console.WriteLine("usage: ReadVersion.exe <README>");
		Environment.Exit(1);
	}

	public static string ReadVersionFromFile(string fileName)
	{
		using (StreamReader sr = new StreamReader(fileName))
		{
			while (sr.Peek() >= 0)
			{
				string line = sr.ReadLine();
				if (line.StartsWith("Version="))
				{
					return line.Split('=')[1];
				}
			}
		}
		return "none";
	}

	public static void Main(string[] args)
	{
		if (args.Length != 1)
			Usage();

		string version = ReadVersionFromFile(args[0]);
		Console.WriteLine("{0}", version);
	}
}

File: filesystem/simple/python/read_version.py
import sys

def readVersion(filename):
	f = open(filename)
	for line in f.xreadlines():
		line = line.strip()
		if line.startswith('Version='):
			return line.split('=')[1]
	return None

if __name__ == '__main__':
	if len(sys.argv) < 2:
		print 'usage: python read_version.py <README>'
		sys.exit(1)

	version = readVersion(sys.argv[1])
	print version

